### DISCUSSION TITLE
Kubernetes 1.27 'failed to update managedFields'

### DISCUSSION THREAD

**Discussion by the4thamigo-uk** [airbytehq/airbyte#32598] [General]
We are seeing these kinds of messages in our logs.

```
"[SHOULD NOT HAPPEN] failed to update managedFields" err="failed to convert new object (airbyte/destination-s3-check-34-1-luwme; /v1, Kind=Pod) to smd typed: .spec.imagePullSecrets: element 0: associative list with keys has an element that omits key field \"name\" (and doesn't have default value)" versionKind="/, Kind=" namespace="airbyte" name="destination-s3-check-34-1-luwme"
```

These are coming from `k8s_control_plane_component`, which is not happy with the format of the `imagePullSecrets` field in this pod manifest. I could be wrong but I believe the code is here : 

https://github.com/airbytehq/airbyte-platform/blame/main/airbyte-commons-worker/src/main/java/io/airbyte/workers/process/KubePodProcess.java#L583-L597

Our pod manifest is being created like this : 

![image](https://github.com/airbytehq/airbyte/assets/7022874/d08b986b-cf52-42f1-a517-52ff2c79d21c)

This does not happen with Kubernetes 1.26

**Comment by marcosmarxm** [comment]
Hello @the4thamigo-uk there are some users who were able to run using Kubernetes 1.27 and 1.28.
Does the problem you're facing is specific to Destination S3? Do you have other connections or connectors working?

**Reply by the4thamigo-uk** [reply]
Hi, thanks for replying. It seems to affect all connectors. Also we see a slightly different error for the orchestrator-repl-job pods :

![image](https://github.com/airbytehq/airbyte/assets/7022874/878d945e-7b29-4a75-a15d-a4d08a788740)

**Comment by XYenon** [comment]
I can reproduce this on v1.25.16 by these steps:

1. Create a deployment with `imagePullSecrets: [{name: ""}]`
2. Edit this deployment's `imagePullSecrets` to `[{name: "some-value"}]`
3. Vice versa

I think it's a bug of Kubernetes.
